sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_text_analytics_client.py

```diff
@@ -239,6 +250,12 @@ def recognize_entities(  # type: ignore
                 cls=kwargs.pop("cls", entities_result),
                 **kwargs
             )
+        except TypeError as error:
```

Comment:
what about

```
string_index_type = kwargs.pop("string_index_type", None)
if self._api_version != "3.0":
     kwargs.update({"stringn_index_type": string_index_type or self._string_code_unit})
if string_index_type and self._api_versoni. == "3.0":
    raise ValueError("'locale' is only available for API version V2_1_PREVIEW and up")
```

This way we won't have to have 2 separate cchecks for type error / value raising. We won'nt really be able to make this a util (we could try looking into it but it might take a whiile / it might not work), but I think this code is clearer and more compacct

Reply:
Actually, I think maybe we can just remove the `except TypeError` block since I'm raising a ValueError in the `_check_string_index_type_arg` function.  I use a similar check to what you're describing here.

Reply:
I removed all of the extra TypeError checks, and verified that the tests should still assert that a ValueError will still be raised if the model version isn't right.